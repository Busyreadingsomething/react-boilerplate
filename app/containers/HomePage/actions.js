import {
  CHANGE_MESSAGE,
  POST_FAIL,
  POST_SUCCESS,
  SUBMIT_MESSAGE,
  RESET,
} from './constants';

export const changeMessage = value => ({
  type: CHANGE_MESSAGE,
  value,
});

export const submitMessage = () => ({
  type: SUBMIT_MESSAGE,
});

export const postSuccess = () => ({
  type: POST_SUCCESS,
});

export const postFail = () => ({
  type: POST_FAIL,
});

export const reset = () => ({
  type: RESET,
});
