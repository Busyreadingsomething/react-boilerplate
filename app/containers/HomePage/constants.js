export const CHANGE_MESSAGE = 'CHANGE_MESSAGE';
export const SUBMIT_MESSAGE = 'SUBMIT_MESSAGE';
export const POST_MESSAGE = 'POST_MESSAGE';
export const POST_FAIL = 'POST_FAIL';
