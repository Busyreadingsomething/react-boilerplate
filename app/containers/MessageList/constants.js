export const LOAD_MESSAGES = 'LOAD_MESSAGES';
export const MESSAGE_SUCCESS = 'MESSAGE_SUCCESS';
export const MESSAGE_FAIL = 'MESSAGE_FAIL';
