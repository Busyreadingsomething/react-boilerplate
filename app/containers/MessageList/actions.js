import { LOAD_MESSAGES, MESSAGE_SUCCESS, MESSAGE_FAIL } from './constants';

export const loadMessages = () => ({
  type: LOAD_MESSAGES,
});

export const messageSuccess = messages => ({
  type: MESSAGE_SUCCESS,
  messages,
});

export const messageFail = () => ({
  type: MESSAGE_FAIL,
});
