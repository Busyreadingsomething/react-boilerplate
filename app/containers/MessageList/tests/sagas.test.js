/**
 * Tests for HomePage sagas
 */

import { put, takeEvery } from 'redux-saga/effects';

import { messageSuccess, messageFail } from '../actions';
import watcherGet, { workerGet } from '../sagas';
import { LOAD_MESSAGES } from '../constants';

/* eslint-disable redux-saga/yield-effects */
describe('workerGet Saga', () => {
  let getListGenerator;
  const expected = {
    data: [{ id: 1, note: 'NOTE' }],
  };
  // We have to test twice, once for a successful load and once for an unsuccessful one
  // so we do all the stuff that happens beforehand automatically in the beforeEach
  beforeEach(() => {
    getListGenerator = workerGet();

    const callDescriptor = getListGenerator.next().value;
    expect(callDescriptor).toMatchSnapshot();
  });

  it('should dispatch the messageSuccess action if it requests fails', () => {
    const putDescriptor = getListGenerator.next(expected).value;
    expect(putDescriptor).toEqual(put(messageSuccess(expected.data)));
  });

  it('should call the repoLoadingError action if the response errors', () => {
    const error = new Error('bad error');
    const putDescriptor = getListGenerator.throw(error).value;
    expect(putDescriptor).toEqual(put(messageFail()));
  });
});

describe('watcherGet Saga', () => {
  const messageDataSaga = watcherGet();

  it('should start task to watch for LOAD_REPOS action', () => {
    const takeEveryDescriptor = messageDataSaga.next().value;
    expect(takeEveryDescriptor).toEqual(takeEvery(LOAD_MESSAGES, workerGet));
  });
});
