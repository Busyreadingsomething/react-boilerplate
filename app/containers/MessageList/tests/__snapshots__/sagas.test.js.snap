// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`workerGet Saga should call the repoLoadingError action if the response errors 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;

exports[`workerGet Saga should dispatch the messageSuccess action if it requests fails 1`] = `
Object {
  "@@redux-saga/IO": true,
  "CALL": Object {
    "args": Array [],
    "context": null,
    "fn": [Function],
  },
}
`;
